src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -118,19 +116,23 @@ public boolean Disconnect(String peerId){
     public boolean  DisconnectIncomingConnections(){
 
         boolean ret = false;
+
         if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                    ret = true;
+            synchronized(mServerSocketList) {
+
+                for (int i = 0; i < mServerSocketList.size(); i++) {
+                    BtToServerSocket tmp = mServerSocketList.get(i);
+                    if (tmp != null) {
+                        print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
+                        tmp.Stop();
+                        tmp = null;
+                        mServerSocketList.set(i, tmp);
+                        ret = true;
+                    }
                 }
-            }
 
-            mServerSocketList.clear();
+                mServerSocketList.clear();
+            }
```

Comment:
@DrJukka Also sorry to stick this comment here but I'm limited by how GitHub reviews work. In any case, why is GetKeyValue and SetKeyValue still around? I thought we were going to get rid of these? Sorry if I've lost my mind on this one but I thought this was functionality we removed?